librz/core/cmd/cmd_print.c

```diff
@@ -7198,3 +7198,674 @@ CMD_PRINT_BYTE_ARRAY_HANDLER_NORMAL(rz_cmd_print_byte_array_swift_handler, RZ_LA
+			disassemble_till_return_is_found(core, core->offset + i, limit, state);
+			break;
+		case RZ_OUTPUT_MODE_STANDARD:
+			rz_cons_printf("[%s0x%08" PFMT64x "%s] %s0x%08" PFMT64x "%s\n", off_color, core->offset + i, end_color, num_color, number, end_color);
+			disassemble_till_return_is_found(core, core->offset + i, limit, state);
+			break;
+		case RZ_OUTPUT_MODE_JSON:
+			pj_o(state->d.pj);
+			pj_kn(state->d.pj, "address", core->offset + i);
+			pj_kn(state->d.pj, "bits", asm_bits);
+			pj_kn(state->d.pj, "word", number);
+			pj_ka(state->d.pj, "opcodes");
+			disassemble_till_return_is_found(core, core->offset + i, limit, state);
+			pj_end(state->d.pj);
+			pj_end(state->d.pj);
+			break;
+		default:
+			rz_warn_if_reached();
+			return RZ_CMD_STATUS_ERROR;
+		}
+	}
+	rz_cmd_state_output_array_end(state);
+
+	return RZ_CMD_STATUS_OK;
+}
+
+static bool core_walk_function_blocks(RzCore *core, RzAnalysisFunction *f, RzCmdStateOutput *state, char type_print, bool fromHere) {
+	RzListIter *iter;
+	RzAnalysisBlock *b = NULL;
+	const char *orig_bb_middle = rz_config_get(core->config, "asm.bb.middle");
+	rz_config_set_i(core->config, "asm.bb.middle", false);
+
+	if (rz_list_length(f->bbs) >= 1) {
+		ut32 fcn_size = rz_analysis_function_realsize(f);
+		b = rz_list_get_top(f->bbs);
+		if (b->size > fcn_size) {
+			b->size = fcn_size;
+		}
+	}
+
+	rz_list_sort(f->bbs, (RzListComparator)bbcmp);
+	if (state->mode == RZ_OUTPUT_MODE_JSON) {
+		rz_list_foreach (f->bbs, iter, b) {
+			ut8 *buf = malloc(b->size);
+			if (!buf) {
+				RZ_LOG_ERROR("cannot allocate %" PFMT64u " byte(s)\n", b->size);
+				return false;
+			}
+			(void)rz_io_read_at(core->io, b->addr, buf, b->size);
+			rz_core_print_disasm_json(core, b->addr, buf, b->size, 0, state->d.pj);
+			free(buf);
+		}
+	} else {
+		bool asm_lines = rz_config_get_i(core->config, "asm.lines.bb");
+		bool emu = rz_config_get_i(core->config, "asm.emu");
+		ut64 saved_gp = 0;
+		ut8 *saved_arena = NULL;
+		int saved_stackptr = core->analysis->stackptr;
+		if (emu) {
+			saved_gp = core->analysis->gp;
+			saved_arena = rz_reg_arena_peek(core->analysis->reg);
+		}
+		rz_config_set_i(core->config, "asm.lines.bb", 0);
+		rz_list_foreach (f->bbs, iter, b) {
+			pr_bb(core, f, b, emu, saved_gp, saved_arena, type_print, fromHere);
+		}
+		if (emu) {
+			core->analysis->gp = saved_gp;
+			if (saved_arena) {
+				rz_reg_arena_poke(core->analysis->reg, saved_arena);
+				RZ_FREE(saved_arena);
+			}
+		}
+		core->analysis->stackptr = saved_stackptr;
+		rz_config_set_i(core->config, "asm.lines.bb", asm_lines);
+	}
+	rz_config_set(core->config, "asm.bb.middle", orig_bb_middle);
+	return true;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_disassemble_recursively_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	RzAnalysisFunction *function = rz_analysis_get_fcn_in(core->analysis, core->offset, 0);
+	// RZ_ANALYSIS_FCN_TYPE_FCN|RZ_ANALYSIS_FCN_TYPE_SYM);
```

Comment:
I think it is worth expanding this comment a bit.

Reply:
these comes from the original code. i do not know why they are like this, so i kept the comment. so much stuff needs to be rewritten